src/main/kotlin/no/fdk/dataset/preview/service/PreviewService.kt

```diff
@@ -39,54 +38,79 @@ class PreviewService(
         }
     }
 
-    @Throws(Exception::class)
-    private fun readCSV(resourceUrl: String, maxNumberOfRows: Int): Table {
-        if(LOGGER.isDebugEnabled) {
-            LOGGER.debug("Reading CSV for $resourceUrl")
+    private fun getMaxNumberOfRows(rows: Int?): Int {
+        if(rows == null) {
+            return DEFAULT_ROWS
         }
+        return if(rows > MAX_ROWS) MAX_ROWS else rows
+    }
+
+    fun readAndParseResource(resourceUrl: String, rows: Int?): Preview {
+        logDebug("Read and parse resource $resourceUrl")
 
-        val body = downloader.download(resourceUrl)
-        if(!isCsv(body.contentType()?.subtype)) {
-            throw Exception("Invalid content type ${body.contentType().toString()}")
+        try {
+            val body = downloader.download(resourceUrl)
+            if(isCsv(body.contentType())) {
+                logDebug("Parsing CSV")
+                val delimiter = detectDelimiter(body.byteStream())
+                logDebug("Detected delimiter $delimiter")
+
+                CSVFormat.DEFAULT.builder()
+                    .setDelimiter(delimiter)
+                    .build()
+                    .parse( InputStreamReader( BOMInputStream(
+                        downloader.download(resourceUrl).byteStream()), StandardCharsets.UTF_8))
```

Comment:
er det ikke mulig å bruke det som ble lastet ned tidligere?
```suggestion
                        body.byteStream()), StandardCharsets.UTF_8))
```

Reply:
Nei den forrige bytestream er closed!

Reply:
Men det fungerer å hente byteStream for plain? snodig

Reply:
Det fungerer siden vi leser byteStream i detectDelimiter for csv :)